plugins/parsers/csv/parser.go

```diff
@@ -0,0 +1,156 @@
+package csv
+
+import (
+	"bytes"
+	"encoding/csv"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+)
+
+type CSVParser struct {
```

Comment:
It doesn't look like the quote character can be customized when using the Go csv parser, and we would want to stick to this implementation.

The other options sound great, but I don't think they are must have and we could add them later depending on available time.

Reply:
I can definitely add features to allow comments and trimming.  I know quote characters aren't supported by the go csv parser.  Regarding column skipping, there is already functionality for that by simply not adding the column name to either `csv_tag_columns` or `csv_field_columns.`  The header row count raises a few issues about how a header with more than one line would be interpreted, unless we decide to skip it entirely.  It would most likely not mesh well with the function to extract column names from the header. We would have to decide how we want to configure that; I think we could probably pair it with the row skipping configuration.